app-libs/stf/src/hash.rs

```diff
@@ -31,3 +34,21 @@ pub enum TrustedOperationOrHash<Hash> {
 	/// Raw extrinsic
 	Operation(TrustedOperation),
 }
+
+impl<Hash> TrustedOperationOrHash<Hash> {
+	pub fn from_top(top: TrustedOperation) -> Self {
+		TrustedOperationOrHash::Operation(top)
+	}
+}
+
+/// Trait to compute a hash of self.
+pub trait Hash<Output> {
+	fn hash(&self) -> Output;
+}
+
+/// Blanket implementation of `Hash` for anything that implements `Decode`.
+impl<T: Encode> Hash<H256> for T {
+	fn hash(&self) -> H256 {
+		blake2_256(&self.encode()).into()
+	}
+}
```

Comment:
Yes, you're right, I should improve this. I switched to a blanket implementation because I thought that's what you'd want 😏  But I'll probably place the trait in a separate crate then, and add a specific implementation for `TrustedOperation`. That also gets rid of the potential wrong usage for `SgxExternalities`.

Reply:
Hahahah, that is correct I usually like blanket implementations, just a bit worried about the ambiguity with for the state, but as we are aware of it, I think it is fine! :D